tree/treeplayer/inc/TBranchProxy.h

```diff
@@ -92,6 +95,8 @@ namespace Detail {
          TLeaf   *fLeafCount;    // eventual auxiliary leaf (for example holding the size)
       };
 
+      std::unique_ptr<TNotifyLink<TBranchProxy>> fNotify; // Callback object used by the TChain to update this proxy
```

Comment:
Is `TNotifyLink<TBranchProxy> fNotify;` enough?

Reply:
maybe ... we also need a trigger/flag for 'has been added to the TTree' ...